### PR TITLE
table: migrate test-infra to testify

### LINKED ISSUE #26865: migrate test-infra to testify for `table`
You may try to analyze first and spawn more subtasks on demand (no strict) instead of one big PR handle the whole package which is hard to review.

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #26865 

### Release note

```release-note
None
```

### CODE REVIEW

File: table/column_test.go [hunk 1]
```diff
@@ -246,64 +250,70 @@ func (t *testTableSuite) TestGetZeroValue(c *C) {
 	}
 	sc := new(stmtctx.StatementContext)
 	for _, tt := range tests {
-		colInfo := &model.ColumnInfo{FieldType: *tt.ft}
-		zv := GetZeroValue(colInfo)
-		c.Assert(zv.Kind(), Equals, tt.value.Kind())
-		cmp, err := zv.CompareDatum(sc, &tt.value)
-		c.Assert(err, IsNil)
-		c.Assert(cmp, Equals, 0)
+		t.Run(fmt.Sprintf("%+v \n", tt.ft), func(t *testing.T) {
```
Comment (tisonkun) [inline table/column_test.go:253]:
```suggestion
		t.Run(fmt.Sprintf("%+v", tt.ft), func(t *testing.T) {
```

can we reduce extra bytes?

File: table/column_test.go [hunk 2]
```diff
@@ -246,64 +250,70 @@ func (t *testTableSuite) TestGetZeroValue(c *C) {
...
 
 	colInfoS := model.ColumnInfo{
 		FieldType: *types.NewFieldType(mysql.TypeString),
 		State:     model.StatePublic,
 	}
 	val, err = CastValue(ctx, types.NewDatum("test"), &colInfoS, false, false)
-	c.Assert(err, IsNil)
-	c.Assert(val, NotNil)
+	require.NoError(t, err)
+
```
Comment (tisonkun) [inline table/column_test.go:287]:
```suggestion
```

extra blank line

File: table/column_test.go [hunk 3]
```diff
@@ -450,25 +460,25 @@ func (t *testTableSuite) TestGetDefaultValue(c *C) {
 		ctx.GetSessionVars().StmtCtx.BadNullAsWarning = !tt.strict
 		val, err := GetColDefaultValue(ctx, tt.colInfo)
 		if err != nil {
-			c.Assert(tt.err, NotNil, Commentf("%v", err))
+			require.Errorf(t, tt.err, "%v", err)
```
Comment (tisonkun) [inline table/column_test.go:463]: weird, shall we assert error message equal?

cc @coocood

File: table/column_test.go [hunk 4]
```diff
@@ -450,25 +460,25 @@ func (t *testTableSuite) TestGetDefaultValue(c *C) {
...
+			require.Equal(t, tt.val, val)
 		}
 	}
 
 	for _, tt := range tests {
 		ctx.GetSessionVars().StmtCtx.BadNullAsWarning = !tt.strict
 		val, err := GetColOriginDefaultValue(ctx, tt.colInfo)
 		if err != nil {
-			c.Assert(tt.err, NotNil, Commentf("%v", err))
+			require.Errorf(t, tt.err, "%v", err)
```
Comment (tisonkun) [inline table/column_test.go:477]: ditto @coocood

**Review by tisonkun — COMMENTED** [review]
Thanks for your contribution @feitian124 ! Comments inline.

File: table/column_test.go (hunk 1 above)
Comment (feitian124) [reply table/column_test.go:253]: i want to make the test logs more clean, i assume an `new line` could be better but need confirm.
will check later.

Comment (feitian124) [reply table/column_test.go]: fixed

File: table/column_test.go (hunk 3 above)
Comment (tisonkun) [reply table/column_test.go:462]: With an offline discussion with @coocood we agree that postpone the problem later and focus on migration in this PR.

File: table/column_test.go (hunk 4 above)
Comment (tisonkun) [reply table/column_test.go:476]: ditto

**Review by tisonkun — APPROVED** [review]
LGTM

File: table/column_test.go (hunk 3 above)
Comment (feitian124) [reply table/column_test.go:462]: :ok_hand:

**Comment by feitian124** [comment]
hi @tisonkun , seems @winoros busy on other things.
could you suggest someone other to help review? thanks.

**Comment by tisonkun** [comment]
@feitian124 get it.

/cc @xhebox @mmyj 

Please take a look >_<

File: table/column_test.go [hunk 5]
```diff
@@ -47,117 +42,126 @@ func (t *testTableSuite) TestString(c *C) {
...
 	col.Flag = mysql.AutoIncrementFlag
 	cols := []*Column{col, col}
 	err := CheckOnce(cols)
-	c.Assert(err, NotNil)
+	require.Error(t, err)
 	cols = cols[:1]
 	err = CheckNotNull(cols, types.MakeDatums(nil))
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
+	require.NoError(t, err)
```
Comment (mmyj) [inline table/column_test.go:114]: duplicated？

Comment (feitian124) [reply table/column_test.go]: sorry, fixed

**Review by mmyj — APPROVED** [review]
lgtm

**Review by xhebox — APPROVED** [review]

**Pull request merged**